src/test/resources/test-suite-data-java.json

```
@@ -0,0 +1,57 @@
+[
+  {
+   "description": "a namespace is required",
+    "purl": "pkg:maven/io@1.3.4",
+    "canonical_purl": "pkg:maven/io@1.3.4",
+    "type": "maven",
+    "namespace": null,
+    "name": null,
+    "version": null,
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": true
+  },
+  {
+    "description": "a namespace is required",
+    "purl": "pkg:maven//io@1.3.4",
+    "canonical_purl": "pkg:maven//io@1.3.4",
+    "type": "maven",
+    "namespace": null,
+    "name": null,
+    "version": null,
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": true
+  },
+  {
+    "description": "valid debian purl containing a plus in the name and version",
+    "purl": "pkg:deb/debian/g++-10@10.2.1+6",
+    "canonical_purl": "pkg:deb/debian/g%2B%2B-10@10.2.1%2B6",
+    "type": "deb",
+    "namespace": "debian",
+    "name": "g++-10",
+    "version": "10.2.1+6",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "Maven Central is too permissive",
+    "purl": "pkg:maven/net.databinder/dispatch-http%252Bjson_2.7.3@0.6.0",
+    "canonical_purl": "pkg:maven/net.databinder/dispatch-http%252Bjson_2.7.3@0.6.0",
+    "type": "maven",
+    "namespace": "net.databinder",
+    "name": "dispatch-http%2Bjson_2.7.3",
+    "version": "0.6.0",
+    "is_invalid": false
+  },
+  {
+  "description": "PURLs are ASCII",
+  "purl": "pkg:nuget/史密斯图wpf控件@1.0.3",
+  "canonical_purl": "pkg:nuget/%E5%8F%B2%E5%AF%86%E6%96%AF%E5%9B%BEwpf%E6%8E%A7%E4%BB%B6@1.0.3",
+  "type": "nuget",
+  "name": "\u53f2\u5bc6\u65af\u56fewpf\u63a7\u4ef6",
+  "version": "1.0.3",
+  "is_invalid": false
+}
```

Correct indentation:
```suggestion
    "description": "PURLs are ASCII",
    "purl": "pkg:nuget/史密斯图wpf控件@1.0.3",
    "canonical_purl": "pkg:nuget/%E5%8F%B2%E5%AF%86%E6%96%AF%E5%9B%BEwpf%E6%8E%A7%E4%BB%B6@1.0.3",
    "type": "nuget",
    "name": "\u53f2\u5bc6\u65af\u56fewpf\u63a7\u4ef6",
    "version": "1.0.3",
    "is_invalid": false
  }
```

---

Should #180 also specify 4 spaces of indentation for JSON?
[Personally I prefer 2 :wink:]

---

I was just going to have it match whatever upstream has and never add tests to that one.

I figured make two files one for data that hasn't yet or won't ever go upstream.

JUnit 5 doesn't have a file source and certainly not a JSON file source but I thought about adding json-params anyway so it will treat each entry in the JSON Array as a separate test.

---

[JUnit Pioneer](https://junit-pioneer.org/docs/json-argument-source/) can read JSON files from the classpath.

I am working to convert most of those tests into parameterized tests.

---

I already have a branch in my local fork for the parameterized tests for the test data ones I think.

JUnit Pioneer requires Java 11. The old Java 8 1.x version may work or it may not but it's not actively developed.

---

My suggestion on the formatting was purely to make it consistent. We should be autoformatting the code. For JSON I prefer 2 spaces as well. In this case, the rest of the file was already 4 spaces.

---

It's fixed and set to 2 spaces which incidentally matches what the upstream data is already using.

Semi-related to this, it was suggested to make the `pom.xml` 2 spaces as well, basically leaving only `*.java` files at 4 spaces and the rest at 2 spaces. Using 2 for `pom.xml` matches the Maven "canonical" pom format, I believe. Maven actually has a canonical ordering as well.